resolver/resolver.go

```diff
@@ -51,8 +47,25 @@ func New(version string, config records.Config) *Resolver {
 		rs:      &records.RecordGenerator{},
 		rng:     rand.New(rand.NewSource(time.Now().UnixNano())),
 	}
-	r.extResolver = r.defaultExtResolver
+
+	timeout := 5 * time.Second
+	if config.Timeout != 0 {
+		timeout = time.Duration(config.Timeout) * time.Second
+	}
+
+	client := &dns.Client{
+		DialTimeout:  timeout,
+		ReadTimeout:  timeout,
+		WriteTimeout: timeout,
+	}
+
+	r.extResolver = exchanger.Decorate(client,
```

Comment:
what do you think about another decorator that accepts a list of resolvers to iterate through? it would get rid of the loop in HandleNonMesos.

this "resolver builder" section could then be wrapped with an `if config.ExternalOn {}` and then in HandleNonMesos the check could if `if r.extResolver != nil {}`. Helps to make config just a little less leaky since HandleNonMesos would no longer reference it.